### ISSUE TITLE
Big Endian support ?

### ISSUE THREAD

**Issue by BeWorld2018** [XProger/OpenLara#397]
Hi, i port some program to MorphOS but platform is BE, opengl 1.2 and no shaders (seem ok for 2 last point)...
I try to compile OpenLara but seem not bigendian compatible.

Have you plan to fix it ? plaform concern : XBox360, PS3, Wii and AmigaOS compatible

Thanks you

**Comment by XProger** [comment]
Yep, it's in plan for new version of engine (fixed), it's already compatible with BE due 3DO support, but the code isn't complete and it uses pre-converted levels format.

**Comment by Kroc** [comment]
Mac PPC folks will also be thankful!

**Comment by XProger** [comment]
@Kroc yep, I've iBook G4 for this reason 8)

**Comment by BeWorld2018** [comment]
Ok, Thank you for your answer.

**Comment by jnmartin84** [comment]
I had to do some work in this area to get OpenLara running on the Nintendo 64.
It isn't quite in shape for a pull request but for reference the changes can be found in my OL fork under the nintendo64 branch:
https://github.com/jnmartin84/OpenLara/tree/nintendo64

**Comment by XProger** [comment]
@jnmartin84 hi, Stream class already has readBE16 and readBE32
According roadmap (#353) the main branch code is deprecated and will be archived after "fixed" branch will be completed. Of course you can continue working on your fork, but the code isn't suitable for old platforms such as N64.

**Comment by jnmartin84** [comment]
Sure, understood.

My original intent was to introduce a compile-time check/define for endian-ness and eventually work through all of the uses of read_/read_LE/read_BE and make them behave appropriately regardless of endian-ness.

I never finished working that through, but I was able to identify all of the code with little-endian assumptions on a big-endian platform. I can finish working that through so there is better platform-independence in the data file loaders, if that is of interest to you/you haven't already done that work for the future.

Whenever your "fixed" branch is mature I can move things over to that.
